lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -468,6 +468,37 @@ func (c *Client) start() {
 						return
 					}
 				}
+			case tdp.SharedDirectoryReadResponse:
+				if c.cfg.AllowDirectorySharing {
+
+					var readData *C.uint8_t
+					if m.ReadDataLength > 0 {
+						readData = (*C.uint8_t)(unsafe.Pointer(&m.ReadData[0]))
+					} else {
+						readData = (*C.uint8_t)(unsafe.Pointer(&m.ReadData))
+					}
```

Comment:
Is the Rust code going to read this data or write to this memory?

Reply:
Rust code is reading this data. This is handler of response for a read call on the Windows side, so we'll read from it and pass the data into Windows using RDP.